src/java/org/apache/cassandra/index/sai/disk/v2/hnsw/DiskBinarySearch.java

```diff
@@ -52,4 +52,32 @@ else if (target > value)
         }
         return -1;
     }
+
+    /**
+     * Similar to searchInt but returns index of a value greater or equal to the target, -1 if not found.
```

Comment:
Can we consolidate and have `searchInt` to return `-x -1`?

Reply:
i'd keep these separate but will refactor